app/controllers/admin/chat_channels_controller.rb

```diff
@@ -29,6 +29,17 @@ def remove_user
       redirect_back(fallback_location: "/admin/chat_channels")
     end
 
+    def destroy
+      @chat_channel = ChatChannel.find(params[:id])
+      if @chat_channel.users.count.zero?
+        @chat_channel.destroy
+        flash[:success] = "Channel was successfully deleted."
+      else
+        flash[:alert] = "Channel NOT deleted, because it still has users."
```

Comment:
**Suggestion (non-blocking)**: Maybe change this to `Channel cannot be deleted, because it still has users`